File: aizu/bubble_sort.cpp
#include <iostream>
using namespace std;
static const int MAX = 100;

int bubbleSort(int A[], int N)
{
    int sw = 0;

    bool flag = 1;
    for (int i = 0; flag; i++)
    {
        flag = 0;
        for (int j = N - 1; j >= i + 1; j--)
        {
            if (A[j] < A[j - 1])
            {
                swap(A[j], A[j - 1]);
                flag = 1;
                sw++;
            }
        }
    }
    return sw;
};

int main()
{
    int A[MAX], j, N, sw;
    cin >> N;
    for (int k = 0; k < N; k++)
    {
        cin >> j;
        A[k] = j;
    }
    sw = bubbleSort(A, N);
    for (int i = 0; i < N; i++)
    {
        if (i)
        {
            cout << " ";
        }
        cout << A[i];
    }
    cout << endl;
    cout << sw << endl;
    return 0;
}

File: Leetcode_solutions/bubble_sort.js
function bubbleSort(A, N) {
  flag = 1;
  while (flag) {
    flag = 0;
    for (let j = N - 1; j > 0; j--) {
      if (A[j] < A[j - 1]) {
        sw = A[j];
        A[j] = A[j - 1];
        A[j - 1] = sw;
        flag = 1;
      }
    }
  }
  return A;
}
